src/analysis/processing/pdal/qgsalgorithmpdaltile.cpp

```diff
@@ -0,0 +1,123 @@
+/***************************************************************************
+                         qgsalgorithmpdaltile.cpp
+                         ---------------------
+    begin                : February 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpdaltile.h"
+
+#include "qgsrunprocess.h"
+#include "qgspointcloudlayer.h"
+
+///@cond PRIVATE
+
+QString QgsPdalTileAlgorithm::name() const
+{
+  return QStringLiteral( "tile" );
+}
+
+QString QgsPdalTileAlgorithm::displayName() const
+{
+  return QObject::tr( "Tile" );
+}
+
+QString QgsPdalTileAlgorithm::group() const
+{
+  return QObject::tr( "Point cloud data management" );
+}
+
+QString QgsPdalTileAlgorithm::groupId() const
+{
+  return QStringLiteral( "pointclouddatamanagement" );
+}
+
+QStringList QgsPdalTileAlgorithm::tags() const
+{
+  return QObject::tr( "export,tile" ).split( ',' );
+}
+
+QString QgsPdalTileAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates tiles from input data." );
+}
+
+QgsPdalTileAlgorithm *QgsPdalTileAlgorithm::createInstance() const
+{
+  return new QgsPdalTileAlgorithm();
+}
+
+void QgsPdalTileAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "LAYERS" ), QObject::tr( "Input layers" ), QgsProcessing::TypePointCloud ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "LENGTH" ), QObject::tr( "Tile length" ), QgsProcessingParameterNumber::Double, 1000.0, false, 1 ) );
+
+  std::unique_ptr< QgsProcessingParameterCrs > paramCrs = std::make_unique< QgsProcessingParameterCrs >( QStringLiteral( "CRS" ), QObject::tr( "Assign CRS" ), QVariant(), true );
+  paramCrs->setFlags( paramCrs->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( paramCrs.release() );
+
+  std::unique_ptr< QgsProcessingParameterFile > paramTempDir = std::make_unique< QgsProcessingParameterFile >( QStringLiteral( "TEMP_DIR" ), QObject::tr( "Temp directory" ), QgsProcessingParameterFile::Folder, QString(), QVariant(), true );
```

Comment:
Why aren't we using the standard processing temp directory here?

Reply:
We are using standard processing temp directory for outputs, but this specific parameter is allows user to defined a location for intermediate files created by `pdal_wrench` during tiling process. By default `pdal_wrench` creates these intermediate files next to the input data, but if this location has limited space or has poor IO performance, one may want to use another location for intermediate files.

Reply:
> but if this location has limited space or has poor IO performance, one may want to use another location for intermediate files.

I'd argue that the same consideration also applies to large rasters and vectors, and that by setting the general processing temp dir location to a appropriate drive/folder once we'd cover all use cases...

Reply:
This is a bit of a special case, because the data can get inflated as much as 10x in the temporary dir (when using compressed LAZ files), and for example in my case I would be connecting an external drive just to serve as a temporary storage - so having an option to specify temporary directory here is quite handy...

Reply:
What about if we had an option in the new panel introduced in https://github.com/qgis/QGIS/pull/52235 where a user can set a temp folder for a particular algorithm? That's a more generic approach to this problem which also helps other pdal/non pdal algorithms.